File: examples/notPrincipal/notPrincipal.ts
import { out } from '../../helper/typescript/typescript_test';
import { Statement } from '../../lib';

function getStatement() {
  return (
    // doc-start
    new Statement.S3()
      .deny()
      .allActions()
      .notPrincipal()
      .forUser('1234567890', 'Bob')
      .onObject('example-bucket', '*')
    // doc-end
  );
}
const s = [getStatement()];
out(s);
//deploy(s, 'access'); //Disabled, because we need valid principals to test

File: examples/notPrincipal/notPrincipal.py
import iam_floyd as statement
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')

s = (
    # doc-start
    statement.S3() \
        .allow() \
        .all_actions() \
        .not_principals() \
        .for_user('1234567890', 'Bob')
        .on_object('example-bucket', '*')
    # doc-end
)

all = [s]
out(all)
deploy(all, type='access')
